src/actions/mod.rs

```diff
@@ -92,6 +98,10 @@ impl ActionHandler {
     pub fn init<O: Output>(&self, init_options: InitializationOptions, out: O) {
         trace!("init: {:?}", init_options);
 
+        if let Err(err) = self.infer_config_defaults(&mut *self.config.lock().unwrap()) {
```

Comment:
So my concern was that this might take a long time to run and block other main thread options, rather than about error handling. Given that we're doing a bunch of disk access, I think it is probably still too long running a task to be on the main thread.